### ISSUE TITLE
Failing types on "returning" methods

### ISSUE THREAD

**Issue by santialbo** [knex/knex#4942]
I recently updated to the latest version of knex and I faced what I think is one of the breaking changes from the release notes:

> Changed data structure from RETURNING operation to be consistent with SELECT;

```
const ids =  await knex.from("table").where(...).update({...}, "id");
```
used to return a `string[]` but now returns a `{id: string}[]`, fair enough.

However some of the overloads still reflect the previous behavior. I managed to fix it with patch-package with the following diff but I'm not 100% this is the correct solution.

Am I on the right track?

```diff
diff --git a/node_modules/knex/types/index.d.ts b/node_modules/knex/types/index.d.ts
index d9bf6e5..b50c579 100644
--- a/node_modules/knex/types/index.d.ts
+++ b/node_modules/knex/types/index.d.ts
@@ -692,7 +692,7 @@ export declare namespace Knex {
     ): QueryBuilder<TRecord, DeferredKeySelection<TRecord, never>[]>;
     insert<
       TKey extends StrKey<ResolveTableType<TRecord>>,
-      TResult2 = DeferredIndex.Augment<
+      TResult2 = DeferredKeySelection.Augment<
         UnwrapArrayMember<TResult>,
         ResolveTableType<TRecord>,
         TKey
@@ -720,7 +720,7 @@ export declare namespace Knex {
     ): QueryBuilder<TRecord, TResult2>;
     insert<
       TKey extends string,
-      TResult2 = DeferredIndex.Augment<
+      TResult2 = DeferredKeySelection.Augment<
         UnwrapArrayMember<TResult>,
         TRecord,
         TKey
@@ -828,7 +828,7 @@ export declare namespace Knex {
     update<
       K1 extends StrKey<ResolveTableType<TRecord, 'update'>>,
       K2 extends StrKey<ResolveTableType<TRecord>>,
-      TResult2 = DeferredIndex.Augment<
+      TResult2 = DeferredKeySelection.Augment<
         UnwrapArrayMember<TResult>,
         ResolveTableType<TRecord>,
         K2
@@ -870,7 +870,7 @@ export declare namespace Knex {
     ): QueryBuilder<TRecord, DeferredKeySelection<TRecord, never>[]>;
     update<
       TKey extends StrKey<ResolveTableType<TRecord>>,
-      TResult2 = DeferredIndex.Augment<
+      TResult2 = DeferredKeySelection.Augment<
         UnwrapArrayMember<TResult>,
         ResolveTableType<TRecord>,
         TKey
@@ -925,7 +925,7 @@ export declare namespace Knex {
     returning(column: '*', options?: DMLOptions): QueryBuilder<TRecord, DeferredKeySelection<TRecord, never>[]>;
     returning<
       TKey extends StrKey<ResolveTableType<TRecord>>,
-      TResult2 = DeferredIndex.Augment<
+      TResult2 = DeferredKeySelection.Augment<
         UnwrapArrayMember<TResult>,
         ResolveTableType<TRecord>,
         TKey
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

**Comment by KristjanTammekivi** [comment]
@lorefnon the patch above looks about right, please take a look

**Comment by lorefnon** [comment]
Yeah, this looks good to me as well.

**Comment by kibertoad** [comment]
@santialbo Could you please create a PR from this?

**Comment by santialbo** [comment]
@kibertoad Opened the PR on https://github.com/knex/knex/pull/5031

**Comment by kibertoad** [comment]
Released in 1.0.4
